sql: refactor PLpgSQL-related types to be more ergonomic

mgartner opened the pull request:
#### plpgsqltree: remove redundant prefix of AST type names

Epic: None

Release note: None

#### plpgsql/parser: remove references to non-existant types

The types `PLpgSQLDatum` and `PLpgSQLScalarVar` do not exist. References
to them in `plpgsql.y` have been removed.

Release note: None

#### opt/optbuilder: alias plpgsqltree as ast

Release note: None

mgartner committed:
plpgsqltree: remove redundant prefix of AST type names

Release note: None

mgartner committed:
opt/optbuilder: alias plpgsqltree as ast

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/108397)
<!-- Reviewable:end -->

DrewKimball reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Thank you, this is _much_ nicer.

> Reviewed 10 of 10 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/108397#-:-NbzzU6S0gS23PnQB-9P:b-gc32d8)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @rharding6373)



<!-- Sent from Reviewable.io -->

rharding6373 reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Nice work!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/108397#-:-Nc462ib1ia3resR7Fwv:bcmw7ss)* status: :shipit: complete! 2 of 0 LGTMs obtained (waiting on @mgartner)



<!-- Sent from Reviewable.io -->

mgartner committed:
plpgsql/parser: remove references to non-existant types

The types `PLpgSQLDatum` and `PLpgSQLScalarVar` do not exist. References
to them in `plpgsql.y` have been removed.

Release note: None

mgartner commented:
TFTRs!

bors r+
